cmd/client/command/common.go

```diff
@@ -182,10 +186,20 @@ func readFromFileOrStdin(specFile string, cmd *cobra.Command) ([]byte, string) {
 		Kind string `yaml:"kind"`
 		Name string `yaml:"name"`
 	}
-	err = yaml.Unmarshal(buff, &spec)
-	if err != nil {
-		ExitWithErrorf("%s failed, invalid spec: %v", cmd.Short, err)
-	}
 
-	return buff, spec.Name
+	yamlDocs := strings.Split(string(buff), yamlSeparator)
```

Comment:
the `yamlSeparator` (defined as `---`) could appear in the middle of a line, so it is not correct to simply split the string with the separator.

Reply:
Good point! I need to dig into the spec or some code. 😉